File: easy/working_experience.hs
import System.Environment (getArgs)
import Data.List.Split (splitOn)
import Data.List (nub)

moy  :: String -> Int
moy s | s == "Jan" = 0
      | s == "Feb" = 1
      | s == "Mar" = 2
      | s == "Apr" = 3
      | s == "May" = 4
      | s == "Jun" = 5
      | s == "Jul" = 6
      | s == "Aug" = 7
      | s == "Sep" = 8
      | s == "Oct" = 9
      | s == "Nov" = 10
      | otherwise  = 11

month  :: String -> Int
month s = 12 * (k - 1990) + moy (take 3 s)
        where k = read (drop 4 s)

workm       :: [String] -> [Int]
workm []     = []
workm (x:xs) = [(month y)..(month z)] ++ workm xs
             where [y, z] = splitOn "-" x

work  :: [String] -> String
work s = show $ div (length . nub $ workm s) 12

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (work . splitOn "; ") $ lines input

File: easy/working_experience.lua
function div(a, b) return math.floor(a/b) end

moy = {Jan=1, Feb=2, Mar=3, Apr=4, May=5, Jun=6,
       Jul=7, Aug=8, Sep=9, Oct=10, Nov=11, Dec=12}
function month(s)
  return 12*(tonumber(s:sub(5, 8))-1990) + moy[s:sub(1, 3)] - 1
end

for line in io.lines(arg[1]) do
  local work, w = {}, 0
  for i in line:gsub("; ", ";"):gmatch("[^;]+") do
    for j = month(i:sub(1, 8)), month(i:sub(10, 17)) do
      work[j] = true
    end
  end
  for _, _ in pairs(work) do w = w + 1 end
  print(div(w, 12))
end
